lib/registry.py

```diff
@@ -557,6 +559,11 @@ def request(self, method, url, statuses={200}, out=None, **kwargs):
          out.start(length)
          for chunk in res.iter_content(ch.HTTP_CHUNK_SIZE):
             out.write(chunk)
+            m.update(chunk)
+         # Check for integrity of downloaded blob by validating its hash digest against expected hash digest
+         if (expdigest is not None):
+            if (expdigest != m.hexdigest()):
+               ch.FATAL("incomplete blob download")
```

Comment:
This error message should be specific about what we know and what we don’t know, and speculation about cause should be labeled clearly. In this case, one possibility would be (don’t take this as a given, think about it some):

```suggestion
               ch.FATAL("registry response has bad digest")
```

Specifically, note that at this level, we don’t know whether it’s a blob being downloaded or something else, and we don’t know why the digest did not match.